python/lsst/daf/butler/butler.py

```diff
@@ -178,6 +178,9 @@ def __reduce__(self):
         """
         return (Butler, (self.config, ))
 
+    def __str__(self):
+        return "Butler(collection='{}', root='{}')".format(self.collection, self.datastore.root)
```

Comment:
I think I'd prefer to use `self.datastore.name` here rather than introduce `root` as a top-level datastore concept.

Reply:
Please let the datastore tell you how it wants to be stringified. I suggest defining a `__str__` for `Datastore` that defaults to returning `self.name`. `root` is not a standard concept for all datastores.

Reply:
I've made this change to implement `__str__` for `Datastore`, and a similar one for `Registry`.  I've also added the registry description to the `Butler` string.